### ISSUE TITLE
java.lang.ClassCastException: io.netty.handler.codec.mqtt.MqttMessage cannot be cast to io.netty.handler.codec.mqtt.MqttPublishMessage

### ISSUE THREAD

**Issue by ManuelB** [moquette-io/moquette#434]
### Expected behavior

MQTTMessageLogger should catch exception during logging.

### Actual behavior

Exception during logging kills client connection 

### Minimal yet complete reproducer code (or URL to code) or complete log file
```
2018-10-30 15:09:34 INFO  LoggingHandler:150 - [id: 0xb7160b2c] REGISTERED
2018-10-30 15:09:34 INFO  LoggingHandler:150 - [id: 0xb7160b2c] BIND: 0.0.0.0/0.0.0.0:9443
2018-10-30 15:09:34 INFO  LoggingHandler:150 - [id: 0xb7160b2c, L:/0:0:0:0:0:0:0:0:9443] ACTIVE
2018-10-30 15:09:34 INFO  Server:192 - Configuring embedded Hazelcast instance
2018-10-30 15:09:34 INFO  Server:195 - There are no Hazelcast intercept handlers. The server won't start a Hazelcast instance.
2018-10-30 15:09:34 INFO  ProtocolProcessorBootstrapper:87 - Initializing messages and sessions stores...
2018-10-30 15:09:34 INFO  ProtocolProcessorBootstrapper:183 - Loading storage class io.moquette.persistence.MemoryStorageService
2018-10-30 15:09:34 INFO  ProtocolProcessorBootstrapper:111 - Configuring message interceptors...
2018-10-30 15:09:34 INFO  BrokerInterceptor:49 - Initializing broker interceptor. InterceptorIds=[]
2018-10-30 15:09:34 INFO  ProtocolProcessorBootstrapper:123 - Initializing subscriptions store...
2018-10-30 15:09:34 INFO  CTrieSubscriptionDirectory:50 - Initializing CTrie
2018-10-30 15:09:34 INFO  CTrieSubscriptionDirectory:55 - Initializing subscriptions store...
2018-10-30 15:09:34 INFO  ProtocolProcessorBootstrapper:127 - Configuring MQTT authenticator...
2018-10-30 15:09:34 INFO  ProtocolProcessorBootstrapper:142 - An io.moquette.spi.impl.security.AcceptAllAuthenticator authenticator instance will be used
2018-10-30 15:09:34 INFO  ProtocolProcessorBootstrapper:145 - Configuring MQTT authorizator...
2018-10-30 15:09:34 INFO  ProtocolProcessorBootstrapper:164 - An io.moquette.spi.impl.security.PermitAllAuthorizator authorizator instance will be used
2018-10-30 15:09:34 INFO  ProtocolProcessorBootstrapper:167 - Initializing connection descriptor store...
2018-10-30 15:09:34 INFO  ProtocolProcessorBootstrapper:170 - Initializing MQTT protocol processor...
2018-10-30 15:09:34 INFO  ProtocolProcessor:185 - Initializing messages publisher...
2018-10-30 15:09:34 INFO  Server:178 - Using default SSL context creator
2018-10-30 15:09:34 INFO  NettyAcceptor:131 - Netty is using NIO
2018-10-30 15:09:34 INFO  NettyAcceptor:196 - Server bound to host=0.0.0.0, port=1883, protocol=TCP MQTT
2018-10-30 15:09:34 INFO  NettyAcceptor:242 - Property websocket_port has been setted to disabled. Websocket MQTT will be disabled
2018-10-30 15:09:34 INFO  DefaultMoquetteSslContextCreator:49 - Checking SSL configuration properties...
2018-10-30 15:09:34 INFO  DefaultMoquetteSslContextCreator:51 - Initializing SSL context. KeystorePath = src/main/resources/netty-keystore.jks.
2018-10-30 15:09:34 INFO  DefaultMoquetteSslContextCreator:79 - Loading keystore. KeystorePath = src/main/resources/netty-keystore.jks.
2018-10-30 15:09:34 WARN  DefaultMoquetteSslContextCreator:121 - No keystore has been found in the bundled resources. Scanning filesystem...
2018-10-30 15:09:34 INFO  DefaultMoquetteSslContextCreator:124 - Loading external keystore. Url = /home/manuel/git/AoclGitMigration/machine-management/netty-keystore/src/main/resources/netty-keystore.jks.
2018-10-30 15:09:34 INFO  DefaultMoquetteSslContextCreator:84 - Initializing key manager...
2018-10-30 15:09:34 INFO  DefaultMoquetteSslContextCreator:100 - Initializing SSL context...
2018-10-30 15:09:34 INFO  DefaultMoquetteSslContextCreator:102 - The SSL context has been initialized successfully.
2018-10-30 15:09:34 INFO  NettyAcceptor:196 - Server bound to host=0.0.0.0, port=8883, protocol=SSL MQTT
2018-10-30 15:09:34 INFO  NettyAcceptor:313 - Property secure_websocket_port has been set to disabled. Secure websocket MQTT will be disabled
2018-10-30 15:09:34 INFO  Server:187 - Moquette server has been started successfully in 191 ms
2018-10-30 15:09:38 INFO  messageLogger:61 - C->B CONNECT <null>
2018-10-30 15:09:38 WARN  SessionsRepository:73 - Session does not exist. CId=de.example.ManagerMQTTClient
2018-10-30 15:09:38 INFO  messageLogger:61 - C<-B CONNACK <de.example.ManagerMQTTClient>
2018-10-30 15:09:38 WARN  SessionsRepository:73 - Session does not exist. CId=de.example.ManagerMQTTClient
2018-10-30 15:09:38 INFO  SessionsRepository:131 - Cleaning session. CId=de.example.ManagerMQTTClient
2018-10-30 15:09:38 INFO  ProtocolProcessor:277 - Connected client <de.example.ManagerMQTTClient> with login <null>
2018-10-30 15:09:38 INFO  messageLogger:65 - C->B SUBSCRIBE <de.example.ManagerMQTTClient> to topics [MqttTopicSubscription[topicFilter=measures/#, qualityOfService=EXACTLY_ONCE]]
2018-10-30 15:09:38 INFO  messageLogger:90 - C<-B SUBACK <de.example.ManagerMQTTClient> packetID <1>, grantedQoses [2]
2018-10-30 15:09:38 INFO  ProtocolProcessor:772 - Client <de.example.ManagerMQTTClient> subscribed to topics
2018-10-30 15:09:38 INFO  messageLogger:65 - C->B SUBSCRIBE <de.example.ManagerMQTTClient> to topics [MqttTopicSubscription[topicFilter=commands/#, qualityOfService=EXACTLY_ONCE]]
2018-10-30 15:09:38 INFO  messageLogger:90 - C<-B SUBACK <de.example.ManagerMQTTClient> packetID <2>, grantedQoses [2]
2018-10-30 15:09:38 INFO  ProtocolProcessor:772 - Client <de.example.ManagerMQTTClient> subscribed to topics
2018-10-30 15:09:42 INFO  messageLogger:61 - C->B CONNECT <null>
2018-10-30 15:09:42 WARN  SessionsRepository:73 - Session does not exist. CId=ccd2c769-2135-456a-ab29-8d6916170942
2018-10-30 15:09:42 INFO  messageLogger:61 - C<-B CONNACK <ccd2c769-2135-456a-ab29-8d6916170942>
2018-10-30 15:09:42 WARN  SessionsRepository:73 - Session does not exist. CId=ccd2c769-2135-456a-ab29-8d6916170942
2018-10-30 15:09:42 INFO  SessionsRepository:131 - Cleaning session. CId=ccd2c769-2135-456a-ab29-8d6916170942
2018-10-30 15:09:42 INFO  ProtocolProcessor:277 - Connected client <ccd2c769-2135-456a-ab29-8d6916170942> with login <null>
2018-10-30 15:09:42 INFO  messageLogger:65 - C->B SUBSCRIBE <ccd2c769-2135-456a-ab29-8d6916170942> to topics [MqttTopicSubscription[topicFilter=commands/ccd2c769-2135-456a-ab29-8d6916170942, qualityOfService=AT_LEAST_ONCE]]
2018-10-30 15:09:42 INFO  messageLogger:90 - C<-B SUBACK <ccd2c769-2135-456a-ab29-8d6916170942> packetID <1>, grantedQoses [1]
2018-10-30 15:09:42 INFO  ProtocolProcessor:772 - Client <ccd2c769-2135-456a-ab29-8d6916170942> subscribed to topics
2018-10-30 15:09:42 INFO  messageLogger:75 - C->B PUBLISH <ccd2c769-2135-456a-ab29-8d6916170942> to topics <measures/ccd2c769-2135-456a-ab29-8d6916170942>
2018-10-30 15:09:42 INFO  ProtocolProcessor:453 - Processing PUBLISH message. CId=ccd2c769-2135-456a-ab29-8d6916170942, topic=measures/ccd2c769-2135-456a-ab29-8d6916170942, messageId=2, qos=EXACTLY_ONCE
2018-10-30 15:09:42 INFO  Qos2PublishHandler:80 - Sending publish message to subscribers CId=ccd2c769-2135-456a-ab29-8d6916170942, topic=measures/ccd2c769-2135-456a-ab29-8d6916170942, messageId=2
2018-10-30 15:09:42 INFO  messageLogger:85 - C<-B PUBREC <ccd2c769-2135-456a-ab29-8d6916170942> packetID <2>
2018-10-30 15:09:42 INFO  messageLogger:85 - C->B PUBREL <ccd2c769-2135-456a-ab29-8d6916170942> packetID <2>
2018-10-30 15:09:42 INFO  Qos2PublishHandler:110 - Processing PUBREL message. CId=ccd2c769-2135-456a-ab29-8d6916170942, messageId=2
2018-10-30 15:09:42 INFO  MessagesPublisher:69 - Sending publish message to subscribers. ClientId=ccd2c769-2135-456a-ab29-8d6916170942, topic=measures/ccd2c769-2135-456a-ab29-8d6916170942, messageId=2
2018-10-30 15:09:42 INFO  PersistentQueueMessageSender:52 - Sending PUBLISH message. MessageId=1, CId=de.example.ManagerMQTTClient, topic=measures/ccd2c769-2135-456a-ab29-8d6916170942
2018-10-30 15:09:42 INFO  messageLogger:85 - C<-B PUBCOMP <ccd2c769-2135-456a-ab29-8d6916170942> packetID <2>
2018-10-30 15:09:42 INFO  messageLogger:75 - C<-B PUBLISH <de.example.ManagerMQTTClient> to topics <measures/ccd2c769-2135-456a-ab29-8d6916170942>
2018-10-30 15:09:42 INFO  messageLogger:85 - C->B PUBREC <de.example.ManagerMQTTClient> packetID <1>
2018-10-30 15:09:42 INFO  messageLogger:85 - C<-B PUBREL <de.example.ManagerMQTTClient> packetID <1>
2018-10-30 15:09:42 INFO  messageLogger:75 - C->B PUBLISH <ccd2c769-2135-456a-ab29-8d6916170942> to topics <measures/ccd2c769-2135-456a-ab29-8d6916170942>
2018-10-30 15:09:42 INFO  ProtocolProcessor:453 - Processing PUBLISH message. CId=ccd2c769-2135-456a-ab29-8d6916170942, topic=measures/ccd2c769-2135-456a-ab29-8d6916170942, messageId=3, qos=EXACTLY_ONCE
2018-10-30 15:09:42 INFO  Qos2PublishHandler:80 - Sending publish message to subscribers CId=ccd2c769-2135-456a-ab29-8d6916170942, topic=measures/ccd2c769-2135-456a-ab29-8d6916170942, messageId=3
2018-10-30 15:09:42 INFO  messageLogger:85 - C<-B PUBREC <ccd2c769-2135-456a-ab29-8d6916170942> packetID <3>
2018-10-30 15:09:42 INFO  messageLogger:85 - C->B PUBREL <ccd2c769-2135-456a-ab29-8d6916170942> packetID <3>
2018-10-30 15:09:42 INFO  Qos2PublishHandler:110 - Processing PUBREL message. CId=ccd2c769-2135-456a-ab29-8d6916170942, messageId=3
2018-10-30 15:09:42 INFO  MessagesPublisher:69 - Sending publish message to subscribers. ClientId=ccd2c769-2135-456a-ab29-8d6916170942, topic=measures/ccd2c769-2135-456a-ab29-8d6916170942, messageId=3
2018-10-30 15:09:42 INFO  PersistentQueueMessageSender:52 - Sending PUBLISH message. MessageId=2, CId=de.example.ManagerMQTTClient, topic=measures/ccd2c769-2135-456a-ab29-8d6916170942
2018-10-30 15:09:42 INFO  messageLogger:75 - C<-B PUBLISH <de.example.ManagerMQTTClient> to topics <measures/ccd2c769-2135-456a-ab29-8d6916170942>
2018-10-30 15:09:42 INFO  messageLogger:85 - C<-B PUBCOMP <ccd2c769-2135-456a-ab29-8d6916170942> packetID <3>
2018-10-30 15:09:42 INFO  messageLogger:85 - C->B PUBREC <de.example.ManagerMQTTClient> packetID <2>
2018-10-30 15:09:42 INFO  messageLogger:85 - C<-B PUBREL <de.example.ManagerMQTTClient> packetID <2>
2018-10-30 15:09:42 ERROR NettyMQTTHandler:108 - Unexpected exception while processing MQTT message. Closing Netty channel. CId=ccd2c769-2135-456a-ab29-8d6916170942
java.lang.ClassCastException: io.netty.handler.codec.mqtt.MqttMessage cannot be cast to io.netty.handler.codec.mqtt.MqttPublishMessage
	at io.moquette.server.netty.metrics.MQTTMessageLogger.logMQTTMessage(MQTTMessageLogger.java:74)
	at io.moquette.server.netty.metrics.MQTTMessageLogger.channelRead(MQTTMessageLogger.java:44)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.moquette.server.netty.metrics.MessageMetricsHandler.channelRead(MessageMetricsHandler.java:50)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:349)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:278)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1429)
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1199)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1243)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:502)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:441)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:278)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.moquette.server.netty.metrics.BytesMetricsHandler.channelRead(BytesMetricsHandler.java:51)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:644)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:579)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:496)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:897)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
2018-10-30 15:09:42 INFO  messageLogger:100 - Channel closed <ccd2c769-2135-456a-ab29-8d6916170942>
2018-10-30 15:09:42 INFO  NettyMQTTHandler:100 - Notifying connection lost event. MqttClientId = ccd2c769-2135-456a-ab29-8d6916170942
2018-10-30 15:09:42 INFO  ProtocolProcessor:673 - Lost connection with client <ccd2c769-2135-456a-ab29-8d6916170942>
2018-10-30 15:09:42 INFO  messageLogger:85 - C->B PUBCOMP <de.example.ManagerMQTTClient> packetID <1>
2018-10-30 15:09:42 INFO  messageLogger:85 - C->B PUBCOMP <de.example.ManagerMQTTClient> packetID <2>
2018-10-30 15:09:47 INFO  messageLogger:61 - C->B CONNECT <null>
2018-10-30 15:09:47 INFO  messageLogger:61 - C<-B CONNACK <ccd2c769-2135-456a-ab29-8d6916170942>
2018-10-30 15:09:47 INFO  SessionsRepository:131 - Cleaning session. CId=ccd2c769-2135-456a-ab29-8d6916170942
2018-10-30 15:09:47 INFO  ProtocolProcessor:277 - Connected client <ccd2c769-2135-456a-ab29-8d6916170942> with login <null>
2018-10-30 15:09:47 INFO  messageLogger:65 - C->B SUBSCRIBE <ccd2c769-2135-456a-ab29-8d6916170942> to topics [MqttTopicSubscription[topicFilter=commands/ccd2c769-2135-456a-ab29-8d6916170942, qualityOfService=AT_LEAST_ONCE]]
2018-10-30 15:09:47 INFO  messageLogger:90 - C<-B SUBACK <ccd2c769-2135-456a-ab29-8d6916170942> packetID <5>, grantedQoses [1]
2018-10-30 15:09:47 INFO  ProtocolProcessor:772 - Client <ccd2c769-2135-456a-ab29-8d6916170942> subscribed to topics
2018-10-30 15:09:52 ERROR NettyMQTTHandler:108 - Unexpected exception while processing MQTT message. Closing Netty channel. CId=ccd2c769-2135-456a-ab29-8d6916170942
java.lang.ClassCastException: io.netty.handler.codec.mqtt.MqttMessage cannot be cast to io.netty.handler.codec.mqtt.MqttPublishMessage
	at io.moquette.server.netty.metrics.MQTTMessageLogger.logMQTTMessage(MQTTMessageLogger.java:74)
	at io.moquette.server.netty.metrics.MQTTMessageLogger.channelRead(MQTTMessageLogger.java:44)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.moquette.server.netty.metrics.MessageMetricsHandler.channelRead(MessageMetricsHandler.java:50)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:349)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:278)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1429)
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1199)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1243)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:502)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:441)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:278)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.moquette.server.netty.metrics.BytesMetricsHandler.channelRead(BytesMetricsHandler.java:51)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:644)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:579)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:496)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:897)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
2018-10-30 15:09:52 INFO  messageLogger:100 - Channel closed <ccd2c769-2135-456a-ab29-8d6916170942>
2018-10-30 15:09:52 INFO  NettyMQTTHandler:100 - Notifying connection lost event. MqttClientId = ccd2c769-2135-456a-ab29-8d6916170942
2018-10-30 15:09:52 INFO  ProtocolProcessor:673 - Lost connection with client <ccd2c769-2135-456a-ab29-8d6916170942>
```

### Moquette MQTT version
0.11

### JVM version (e.g. `java -version`)
openjdk version "1.8.0_181"
OpenJDK Runtime Environment (build 1.8.0_181-8u181-b13-0ubuntu0.18.04.1-b13)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)

### OS version (e.g. `uname -a`)
Linux manuel-XPS-13-9360 4.15.0-36-generic #39-Ubuntu SMP Mon Sep 24 16:19:09 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

**Comment by ManuelB** [comment]
This is a duplicate of: #375 and fixed with #389

Solution:

```
properties.put(BrokerConstants.NETTY_MAX_BYTES_PROPERTY_NAME, new Integer(1024*1024*100).toString());
```

**Comment by ManuelB** [comment]
The above erro happens when using the following maven dependencies together:

```
                <dependency>
			<groupId>io.netty</groupId>
			<artifactId>netty-all</artifactId>
			<version>4.1.30.Final</version>
		</dependency>
		<dependency>
			<groupId>io.moquette</groupId>
			<artifactId>moquette-broker</artifactId>
			<version>0.11</version>
		</dependency>
```
